packages/react/src/components/Panel/index.test.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Panel, { PanelHeader, PanelContent } from './';
+import axe from '../../axe';
+
+test('should render the default heading level', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H2');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render a custom heading level', () => {
+  render(<Panel heading={{ text: 'Title', level: 3 }}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H3');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render the label the panel by the heading by default', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
```

Comment:
Very minor style nit. (and for all other <code>{`Title`}</code> ones.

```suggestion
  render(<Panel heading="Title">Content</Panel>);
```